pkg/kubelet/dockertools/exec.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dockertools
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"sync"
+	"time"
+
+	kubecontainer "github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/container"
+	"github.com/fsouza/go-dockerclient"
+	"github.com/golang/glog"
+)
+
+// DockerExecHandler knows how to execute a command in a running Docker container.
+type DockerExecHandler func(client DockerInterface, container *docker.Container, cmd []string, stdin io.Reader, stdout, stderr io.WriteCloser, tty bool) error
+
+var handlers map[string]DockerExecHandler = make(map[string]DockerExecHandler)
+var lock sync.Mutex
+
+// RegisterExecHandler registers a DockerExecHandler.
+func RegisterExecHandler(name string, h DockerExecHandler) {
+	lock.Lock()
```

Comment:
@smarterclayton the reason I did it this way is because the `DockerManager` isn't exposed all the way up to the `KubeletConfig` and I need to allow OpenShift to specify a custom exec handler. If you'd prefer, I can make `DockerExecHandler` and interface and add it to the `KubeletConfig`.